core/src/apps/cardano/get_native_script_hash.py

```diff
@@ -0,0 +1,29 @@
+from trezor import log, wire
+from trezor.enums import CardanoNativeScriptHashDisplayFormat
+from trezor.messages import CardanoNativeScriptHash
+
+from . import native_script, seed
+from .layout import show_native_script, show_script_hash
+
+if False:
+    from trezor.messages import CardanoGetNativeScriptHash
+
+
+@seed.with_keychain
+async def get_native_script_hash(
+    ctx: wire.Context, msg: CardanoGetNativeScriptHash, keychain: seed.Keychain
+) -> CardanoNativeScriptHash:
+    native_script.validate_native_script(msg.script)
+
+    try:
+        script_hash = native_script.get_native_script_hash(keychain, msg.script)
+    except ValueError as e:
```

Comment:
Yep, this can't happen.

Removed in https://github.com/trezor/trezor-firmware/pull/1772/commits/6819215aa240da1e8f3e7749c8a430dd191198d9#diff-f30f1d52cd7d1e04741619751f2a7370eba8f56597c79a8922634ddd5d2c8778R18